lib/net/pool.js

```diff
@@ -862,7 +862,8 @@ class Pool extends EventEmitter {
 
       items.push(node.hash);
 
-      if (items.length === 50000)
+      // Don't send oversized getdata request
+      if (items.length === 27000)
```

Comment:
Where is this number coming from?

Reply:
this will request `getdata` for blocks.
each item is `36 * item.length + sizeVarint(item.length)`.
So it should be able to fit 27777 block requests. I just used 27k. : )